src/test/java/io/confluent/kafkarest/mock/MockTime.java

```diff
@@ -40,6 +47,10 @@ public void sleep(long ms) {
 
   @Override
   public void waitOn(Object on, long ms) throws InterruptedException {
+    // Introduced a thread.sleep to allow context switching in consumer worker.
+    // Without this the worker appears to get stuck in a busy loop and actually
+    // takes longer  to process the messages...
+    Thread.sleep(1);
```

Comment:
I'm not sure I understand what this is supposed to accomplish. Context switching doesn't rely on `Thread.sleep()` being called.